exercises/triangle/description.md

```diff
@@ -2,6 +2,16 @@ The program should raise an error if the triangle cannot exist.
 
 ## Hint
 
-The sum of the lengths of any two sides of a triangle always exceeds or 
-is equal to the length of the third side, a principle known as the _triangle
+In mathematics, a degenerate case is a limiting case in which an element of a
+class of objects is qualitatively different from the rest of the class and hence
+belongs to another, usually simpler, class.
+
+1. If the sum of the lengths of any two sides of a triangle is equal to the
+   length of the third side, there exists a degenerate triangle.
+
+2. The sum of the lengths of any two sides of a triangle always exceeds or is
```

Comment:
Is there a chance that it will be unclear to a reader that the two classes of triangles mentioned (not degenerate vs degenerate) are a separate classification from the one resulting from the lengths of the sides (equilateral, isosceles, scalene)? If there is, we should try to find a way to make the distinction clearer. But if others find it understandable, I think this is good.

I think the last thing is: After a student has read this, the student might think "OK, I have just learned about degenerate triangles, and how should this knowledge apply to this exercise?" and we should give an answer to that potential question in the readme (that they don't violate the triangle inequality - but we won't test for them)